src/OpenLoco/CMakeLists.txt

```diff
@@ -0,0 +1,651 @@
+include(FindGit)
+
+find_package(Git)
+if (Git_FOUND)
+    # Describe current version in terms of closest tag
+    execute_process(
+        COMMAND ${GIT_EXECUTABLE} describe HEAD --always
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        OUTPUT_VARIABLE OPENLOCO_VERSION_TAG
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        # ERROR_QUIET
+    )
+
+    if (${OPENLOCO_VERSION_TAG})
+        # Use a platform agnostic sed equivalent to strip the commit hash from the version tag
+        # i.e. "v22.10-9-g8ff1d207" becomes "v22.10-9 "
+        string(REGEX REPLACE "-g.+$" " " OPENLOCO_VERSION_TAG ${OPENLOCO_VERSION_TAG})
+        # Has to be in two bits due to empty string not being possible to pass to REGEX REPLACE
+        string(STRIP ${OPENLOCO_VERSION_TAG} OPENLOCO_VERSION_TAG)
+    else()
+        # If a low fetch depth is used then nearest tag for git will fail
+        # so just use the project version in that case.
+        set(OPENLOCO_VERSION_TAG ${PROJECT_VERSION})
+    endif()
+    # Define current git branch
+    execute_process(
+        COMMAND ${GIT_EXECUTABLE} rev-parse --abbrev-ref HEAD
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        OUTPUT_VARIABLE OPENLOCO_BRANCH
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_QUIET
+    )
+
+    # Define short commit hash
+    execute_process(
+        COMMAND ${GIT_EXECUTABLE} rev-parse --short HEAD
+        WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+        OUTPUT_VARIABLE OPENLOCO_COMMIT_SHA1_SHORT
+        OUTPUT_STRIP_TRAILING_WHITESPACE
+        ERROR_QUIET
+    )
+
+    # To avoid unnecessary rebuilds, set the current branch and short sha1 hash
+    # only for the one file that uses these definitions: Version.cpp
+    set_property(SOURCE "${CMAKE_CURRENT_SOURCE_DIR}/src/Version.cpp"
+        PROPERTY COMPILE_DEFINITIONS
+        OPENLOCO_VERSION_TAG="${OPENLOCO_VERSION_TAG}"
+        OPENLOCO_BRANCH="${OPENLOCO_BRANCH}"
+        OPENLOCO_COMMIT_SHA1_SHORT="${OPENLOCO_COMMIT_SHA1_SHORT}")
+endif()
+
+# add source files Generated originally with: find src -type f -follow -name '*.mm' -exec echo "\"{}\"" \;
```

Comment:
That comment looks odd: it mentions `.mm`, but all following files are `.cpp`. Is that meant to meant to be a placeholder? `-exec echo` seems redundant when there is no `${CMAKE_CURRENT_SOURCE_DIR}` included

Reply:
It was just a note for reproducing this if required. I manually added the `${CMAKE_CURRENT_SOURCE_DIR}` afterwards as I changed my mind on how best to show it.